### ISSUE TITLE
[BUG] column++ unary operations drop validity data

### ISSUE THREAD

**Issue by revans2** [rapidsai/cudf#3534]
**Describe the bug**
All unary ops allocate a validity buffer, but never populate it.

**Steps/Code to reproduce bug**
Run a unary op with at least one null.  The returned validity buffer is uninitialized.

I wrote a quick patch that should **NOT** be used to verify that it fixes the issue.  Note that I have been using the java API for testing where I found this issue trying to port the java API to column++.

```patch
diff --git a/cpp/src/unary/unary_ops.cuh b/cpp/src/unary/unary_ops.cuh
index ccf4f592c..39d479e10 100644
--- a/cpp/src/unary/unary_ops.cuh
+++ b/cpp/src/unary/unary_ops.cuh
@@ -58,6 +58,12 @@ struct launcher {
         CUDF_EXPECTS(input.size() > 0,                   "Launcher requires input size to be non-zero.");
         CUDF_EXPECTS(input.size() == output_view.size(), "Launcher requires input and output size to be equal.");
 
+        if (input.nullable()) {
+            // Need to copy the validity vector over...
+            //TODO check result
+            cudaMemcpyAsync(output_view.null_mask(), input.null_mask(), (input.size() + 7)/8,cudaMemcpyDeviceToDevice, 0);
+        }
+
         thrust::transform(
             rmm::exec_policy(stream)->on(stream),
             input.begin<T>(),
```

**Comment by jrhemstad** [comment]
@codereport can you take care of this? Should be an easy fix to just copy the mask from input to output. I'd do it like this:

```
output.set_null_mask( device_buffer{input.null_mask(), bitmask_allocation_size_bytes(input.size()));
```

**Comment by codereport** [comment]
@jrhemstad Yes, will fix ASAP

**Comment by harrism** [comment]
Fixed by #3540
